src/main/scala/net/cakesolutions/CakeStandardsPlugin.scala

```diff
@@ -46,6 +46,8 @@ object CakeStandardsPlugin extends AutoPlugin {
             "-Xlint",
             "-Yno-adapted-args",
             "-Ywarn-dead-code",
+            "-Ywarn-value-discard",
```

Comment:
I don't recall false positives with scalatest matchers? I note that when I look at some sample testing code (with warn-value-discard enabled and multiple asserts), I'm not seeing anything untoward - however, I'd need to look at more complex testing code in order to have a higher degree of confidence here.

Current rationale regarding this option in test scopes is as follows:
* on local dev cycles, warnings remain as warnings, and so the developer can always ignore these (but be appraised of them)
* in CI, warnings are errors, and so the developer is forced to deal with these (which can be annoying)
* any value discard (even in testing) should be considered a potential issue, and so needs to be flagged up and provided with visibility in code reviews (e.g. by wrapping the discard action)
* testing has the potential to introduce lots of cases impacting warn-value-discard, and so for some projects it may be more beneficial to disable this option
* disabling warn-value-discard for testing by default probably means it will never be switched on again (e.g. on projects with low warn-value-discard noise).

My understanding of consensus here is as follows:
* enable this setting on testing scopes by default
* on those projects where the settings use is troublesome, then they should disable the setting.

Regardless of decisions here, given the need to switch this setting on and off, I need to ensure that this may be done and so will add in appropriate testing for this to this PR. I'll also try and use the testing scenario to prove that the scalatest concern does not exist (or maybe it does!).